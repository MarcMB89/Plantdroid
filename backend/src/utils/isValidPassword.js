function isValidPassword(password) {
  return password === this.password;
}
module.exports = { isValidPassword };
