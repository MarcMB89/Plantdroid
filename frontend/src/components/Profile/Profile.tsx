import React from 'react';

import {StatusBar, Text, ImageBackground, View, Image, SafeAreaView} from 'react-native';

import {Button} from 'react-native-paper';

import styles from './profileStyle';


export default function Profile({navigation}: any) {
  const onPressHandlerLogout = () => {
    navigation.navigate('Login');
  };
  return (
    <SafeAreaView style={styles.profile_container}>
      <StatusBar backgroundColor="#7BA891" />
      <View style={styles.profile_background}>
        <ImageBackground
          style={styles.profile_background}
          source={{uri: 'https://i.ibb.co/99S3gsz/Background-Plantdroid.jpg'}}>
          <View style={styles.profile_body}>
            <Image
              source={{uri:'https://lh3.googleusercontent.com/a/AATXAJwEEYi9UmJfqRywEb_Krvyoco5a4rdbiZqp-cr2=s360-p-rw-no-mo'}}
              style={styles.profile_photo}
            />
          </View>
                <View style={styles.profile_userData_container}>
                  <Text style={styles.profile_userData}>User </Text>
                  <Text style={styles.profile_userData_text}>Marc Mart√≠n</Text>
                </View>
                <View style={styles.profile_userData_container}>
                    <Text style={styles.profile_userData}>Email </Text>
                    <Text style={styles.profile_userData_text}>marcmartinbarrios@gmail.com</Text>
                </View>
                <Button
                  style={styles.profile_logoutButton}
                  color="#7BA891"
                  mode="contained"
                  onPress={onPressHandlerLogout}>
                  <Text>Logout</Text>
                </Button>
        </ImageBackground>
      </View>
    </SafeAreaView>
  );
}
