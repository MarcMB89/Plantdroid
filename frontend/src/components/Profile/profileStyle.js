import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({

    profile_fixToText: {
        marginTop: 35,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    
    profile_loginButton: {
        marginLeft: 35
    },

    profile_logo: {
        width: 130,
        height: 130,
        borderRadius: 50,
        borderColor: 'black'
    },
    
    profile_body: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 50,
    },

    profile_background: {
        flex: 1,
        backgroundColor: '#dcedc8'
    },

    profile_title: {
        fontSize: 40,
        fontFamily: 'monospace',
        fontWeight: 'bold',
        margin: 10,
        marginTop: 25,
        marginBottom: 11,
        backgroundColor: 'white'
    },

    profile_userData: {
        width: 250,
        height: 65,
        marginBottom: 5,
        marginTop: 5
    },
      
    profile_container: {
        flex: 1,
    },

    profile_body: {
        marginTop: 70,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
      },
    
      profile_photo: {
          width: 130,
          height: 130,
          borderRadius: 100,
          margin: 10,
      },
    
      profile_userData_container: {
        marginLeft: 20,
        marginRight: 100,
        marginTop: 15,
      },
    
      profile_userData: {
        marginLeft: 5,
        marginBottom: 3,
        fontWeight: 'bold',
        fontSize: 25,
      },
    
      profile_userData_text: {
        marginLeft: 5,
        fontSize: 20,
      },
    
      profile_logoutButton: {
        marginTop: 30,
        margin: 5,
      },

})

export default styles;