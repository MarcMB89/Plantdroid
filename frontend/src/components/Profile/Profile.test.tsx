import React from 'react';
import {render} from '../../utils/test.utils';
import Profile from './Profile';

let navigation: any = {
  navigate: jest.fn(),
};

describe('Given a Profile Component', () => {
  describe('When it is rendered', () => {
    let screen: any;
    beforeEach(() => {
      screen = render(<Profile navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});

navigation = {navigation};
