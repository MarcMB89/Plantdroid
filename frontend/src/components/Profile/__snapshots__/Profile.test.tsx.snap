// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Profile Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://i.ibb.co/99S3gsz/Background-Plantdroid.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginTop": 70,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://lh3.googleusercontent.com/a/AATXAJwEEYi9UmJfqRywEb_Krvyoco5a4rdbiZqp-cr2=s360-p-rw-no-mo",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "height": 130,
              "margin": 10,
              "width": 130,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
            "marginRight": 100,
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "marginBottom": 3,
              "marginLeft": 5,
            }
          }
        >
          User 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": 5,
            }
          }
        >
          Marc Mart√≠n
        </Text>
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
            "marginRight": 100,
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "marginBottom": 3,
              "marginLeft": 5,
            }
          }
        >
          Email 
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": 5,
            }
          }
        >
          marcmartinbarrios@gmail.com
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#7BA891",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "margin": 5,
            "marginTop": 30,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text>
                Logout
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
